tests/src/test/scala/quasiquotes/QuasiquoteSuite.scala

```diff
@@ -276,16 +313,65 @@ class QuasiquoteSuite extends FunSuite {
     assert(expr2.show[Syntax] === "bar")
   }
 
-//  test("q\"for (..$enumerators) $expr\"") {
-//    val q"for (..$enumerators) $expr" = q"for (x <- xs; y <- ys) foo(x, y)" // fixme test is broken, so even does not compile
-//  }
+  test("1 q\"for (..$enumerators) $expr\"") {
+    val q"for ($enum1; ..$enumerators; if $cond; $enum2) $exprr" = q"for (a <- as; x <- xs; y <- ys; if bar; b <- bs) foo(x, y)"
+    assert(enumerators.toString === "List(x <- xs, y <- ys)")
+    assert(enumerators(0).show[Syntax] === "x <- xs")
+    assert(enumerators(1).show[Syntax] === "y <- ys")
+    assert(cond.show[Syntax] === "bar")
+    assert(enum1.show[Syntax] === "a <- as")
+    assert(enum2.show[Syntax] === "b <- bs")
+    assert(exprr.show[Syntax] === "foo(x, y)")
+  }
+
+  test("2 q\"for (..$enumerators) $expr\"") {
+    val a = enumerator"a <- as"
+    val b = enumerator"b <- bs"
+    val ab = List(a,b)
+    assert(q"for (..$ab) foo".show[Syntax] === "for (a <- as; b <- bs) foo")
+  }
 
-//  test("q\"for (..$enumerators) yield $expr\"") {
-//    val q"for (..$enumerators) yield $expr" =  q"for (x <- xs; y <- ys) yield foo(x, y)" // fixme test is broken, so even does not compile
+//  test("3 q\"for (..$enumerators) $expr\"") {
+//    val q"for (a <- as; if $cond; ..$enums) bar" = q"for (a <- as; if foo; b <- bs) bar" // fixme does not compile
 //  }
```

Comment:
Could you link https://github.com/scalameta/scalameta/issues/203 in a comment to this test?